ext/gctrack/gctrack.c

```diff
@@ -40,6 +42,26 @@ add_gc_cycle(uint64_t duration)
   }
 }
 
+static void
+add_allocations()
+{
+  record_t *record = last_record;
+  while (record) {
+    record->allocations += 1;
+    record = record->parent;
+  }
+}
+
+static void
+add_context_switch()
+{
+  record_t *record = last_record;
+  while (record) {
```

Comment:
You don't need to do that this aggressively, you can roll up on `#end_record` invoke. 
Might want to look into how this is done here: https://github.com/Shopify/gctrack/blob/master/ext/gctrack/gctrack.c#L149-L152
I changed this a few month back.

Reply:
Ah, that makes a lot more sense. I think I did this back in January before that patch maybe.